tests/src/lib.rs

```diff
@@ -155,3 +164,22 @@ pub fn convert_from_stark_scalar(
     let bytes = x.to_bytes_be();
     generic_ec::Scalar::from_be_bytes(bytes).context("Can't read bytes")
 }
+
+#[cfg(feature = "hd-wallets")]
+pub fn random_derivation_path<E: Curve>(
+    rng: &mut impl rand::RngCore,
+    epub: &cggmp21::slip_10::ExtendedPublicKey<E>,
+) -> (Vec<u32>, generic_ec::Point<E>) {
```

Comment:
I would feel more confident in tests if this function didn't return public key, and instead it would be derived close to its usage place. Not sure about this thought

Reply:
It's only used for signature verification in the end of the tests

Reply:
Well, actually, maybe it's not a bad idea, let me try